streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -369,6 +369,8 @@ public synchronized <K, V> KTable<K, V> table(final String topic,
      * <p>
      * Note that {@link GlobalKTable} always applies {@code "auto.offset.reset"} strategy {@code "earliest"}
      * regardless of the specified value in {@link StreamsConfig} or {@link Consumed}.
+     * Furthermore, {@link GlobalKTable} cannot be a {@link org.apache.kafka.streams.state.VersionedBytesStoreSupplier
```

Comment:
Does it make more sense to point to VersionedBytesStoreSupplier in these docs notes or VersionedKeyValueStore? I see the merits of both, might be inconsistent right now -- the other docs I added, e.g., for TopologyTestDriver link to VersionedKeyValueStore, but perhaps that example is different since it's for getting the store rather than supplying one.

Reply:
Good question -- was only not sure -- I picked `VersionedBytesStoreSupplier` because this is the class they would pass into `Materialized`. -- I think it's ok to have it "inconsistent" as it's a different context. Also easy to change if people get confused.